### ISSUE TITLE
Enum SourceType / Missing customVector case

### ISSUE THREAD

**Issue by NasH5169** [mapbox/mapbox-maps-ios#1399]
Hi,

Following the previous ticket #871 you've fixed the issue by adding the customVector case:

Commit https://github.com/mapbox/mapbox-maps-ios/pull/1080/commits/7cc4b8077cabd306a2c64445241da7d059706fc5

But since the version 10.4.0 the Sources.swift has been renamed by SourceType.swift and the customVector case is not defined in this new class which lead to the same original issue #871 

Thanks.

**Comment by Adraesh** [comment]
I confirm the same issue.

**Comment by NasH5169** [comment]
I really don't understand your answer.

When I am trying to add the CustomGeometrySource, I have this log:

`
2022-06-23 20:26:01.704618+0200 app[11169:928811] [Mapbox] [Error, maps-ios/Example]: Failed to create SourceType from customvector
`

Is there a working sample somewhere?

**Comment by evil159** [comment]
Hi, @NasH5169, @Adraesh the error warning that you see does not indicate an error with the custom geometry source that you added. The log message `Failed to create...` appears when accessing the `Style.allSourceIdentifiers`property, it means that the CustomGeometrySource does not have a corresponding `SourceType` case and won't be represented in the list of all source identifiers. 
You can safely ignore the warning and interact with this source through the dedicated API:
```swift
Style.addCustomGeometrySource(withId:options:)
Style.setCustomGeometrySourceTileData(forSourceId:tileId:features)
Style.invalidateCustomGeometrySourceTile(forSourceId:tileId)
Style.invalidateCustomGeometrySourceRegion(forSourceId:bounds:)
```

Sorry for the confusion this have caused, we will be looking into improving the design of that component.
